source/Applications/Advanced/TransformPointCloudViaArucoMarker/TransformPointCloudViaArucoMarker.cpp

```diff
@@ -0,0 +1,313 @@
+            // f(x,y1) = (x2 - x)/(x2 - x1) * f(q11) + (x - x1)/(x2 - x1) * f(q21)
+            const cv::Point3f fxy1{ (xRoundedUp - x) / (xRoundedUp - xRoundedDown) * q11.x
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q21.x,
+                                    (xRoundedUp - x) / (xRoundedUp - xRoundedDown) * q11.y
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q21.y,
+                                    (xRoundedUp - x) / (xRoundedUp - xRoundedDown) * q11.z
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q21.z };
+            // f(x,y2) = (x2 - x)/(x2 - x1) * f(q12) + (x - x1)/(x2 - x1) * f(q22)
+            const cv::Point3f fxy2{ (xRoundedUp - x) / (xRoundedUp - xRoundedDown) * q12.x
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q22.x,
+                                    (xRoundedUp - x) / (xRoundedUp - xRoundedDown) * q12.y
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q22.y,
+                                    (xRoundedUp - x) / (xRoundedUp - xRoundedDown) * q12.z
+                                        + (x - xRoundedDown) / (xRoundedUp - xRoundedDown) * q22.z };
+
+            // Linear interpolation in y direction
+            // f(x,y) = (y2 - y)/(y2 - y1) * f(x,y1) + (y - y1)/(y2 - y1) * f(x,y2)
+            markerPoints3D.emplace_back(cv::Point3f{ (yRoundedUp - y) / (yRoundedUp - yRoundedDown) * fxy1.x
+                                                         + (y - yRoundedDown) / (yRoundedUp - yRoundedDown) * fxy2.x,
+                                                     (yRoundedUp - y) / (yRoundedUp - yRoundedDown) * fxy1.y
+                                                         + (y - yRoundedDown) / (yRoundedUp - yRoundedDown) * fxy2.y,
+                                                     (yRoundedUp - y) / (yRoundedUp - yRoundedDown) * fxy1.z
+                                                         + (y - yRoundedDown) / (yRoundedUp - yRoundedDown) * fxy2.z });
+        }
+
+        return markerPoints3D;
+    }
+
+    Zivid::Matrix4x4 transformationMatrix(const cv::Matx33d &rotationMatrix, const cv::Vec3d &translationVector)
+    {
+        auto transformMatrix = Zivid::Matrix4x4{ 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 };
+
+        for(size_t row = 0; row < rotationMatrix.rows; row++)
+        {
+            for(size_t col = 0; col < rotationMatrix.cols; col++)
+            {
+                transformMatrix(row, col) = rotationMatrix(row, col);
+            }
+        }
+
+        transformMatrix(0, 3) = translationVector[0];
+        transformMatrix(1, 3) = translationVector[1];
+        transformMatrix(2, 3) = translationVector[2];
+
+        return transformMatrix;
+    }
+
+    Zivid::Matrix4x4 estimateArUcoMarkerPose(const Zivid::PointCloud &pointCloud,
```

Comment:
This seems like an algorithm that could easily be adapted to "estimateCheckarboardPose" instead. Is there a reason why you went for the Aruco?

Reply:
We went for ArUco because people often use ArUco anyway. The markers can be made small and added to a scene permanently for use cases where the accuracy of detection is not too important (e.g. 3D ROI cropping, etc.)\

I also think the checkerboard pose estimation is useful and we might make a similar sample later for the checkerboard, or even better we might get it from the SDK :)